src/Templates/ClassicTemplatesCompatibility.php

```diff
@@ -64,9 +64,14 @@ public function set_filterable_product_data() {
 	 * This method passes the value `is_rendering_php_template` to the front-end of Classic themes,
 	 * so that widget product filter blocks are aware of how to filter the products.
 	 *
+	 * This data only matters on WooCommerce product archive pages.
+	 * On non-archive pages the merchant could be using the All Products block which is not a PHP template.
+	 *
 	 * @return void
 	 */
 	public function set_php_template_data() {
```

Comment:
This method will only get executed on Classic Themes such as Storefront: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/Templates/ClassicTemplatesCompatibility.php#L36-L38